src/node_v8.cc

```diff
@@ -101,7 +98,7 @@ void UpdateHeapSpaceStatisticsBuffer(const FunctionCallbackInfo<Value>& args) {
   Isolate* const isolate = env->isolate();
   double* buffer = env->heap_space_statistics_buffer();
 
-  for (size_t i = 0; i < number_of_heap_spaces; i++) {
+  for (size_t i = 0; i < isolate->NumberOfHeapSpaces(); i++) {
```

Comment:
Suggestion: cache the value in a local so it's only one function call instead of `n` function calls.